permission_handler_android/android/src/main/java/com/baseflow/permissionhandler/PermissionManager.java

```diff
@@ -77,14 +74,20 @@ public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
             return false;
         }
 
-        int status = resultCode == Activity.RESULT_OK
-            ? PermissionConstants.PERMISSION_STATUS_GRANTED
-            : PermissionConstants.PERMISSION_STATUS_DENIED;
-
-        int permission;
+        int status, permission;
 
         if (requestCode == PermissionConstants.PERMISSION_CODE_IGNORE_BATTERY_OPTIMIZATIONS) {
             permission = PermissionConstants.PERMISSION_GROUP_IGNORE_BATTERY_OPTIMIZATIONS;
+
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+                return false;
```

Comment:
Do we want to report `false` or `true` when the device runs a version below Marshmallow? Doze and App Stand-by were not introduced until Marshmallow. It might make sense to return this as `true` from a developer's perspective, as to indicate that the application is not subjected to battery optimizations.

Reply:
That is a good point, I will update the code

Reply:
This is probably also the case for the logic that handles just checking the status of the permission